Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -2325,6 +2339,52 @@ void CGame::Packet_Command(CCommandPacket& Packet)
     }
 }
 
+void CGame::Packet_ObjectBreak(CObjectBreakPacket& Packet)
+{
+    CPlayer* pPlayer = Packet.GetSourcePlayer();
+    if (!pPlayer || !pPlayer->IsJoined())
+        return;
+
+    CObject* pObject = static_cast<CObject*>(CElementIDs::GetElement(Packet.m_ObjectID));
```

Comment:
In this case the cast to CObject is redundant because none of its fields are used. Additionally, since a packet could be compromised the cast without an element type validation may potentially be dangerous.